arm-machinelearning/2016-05-01-preview/swagger/commitmentPlans.json

```diff
@@ -0,0 +1,903 @@
+{
+    "swagger": "2.0",
+    "info": {
+        "version": "v1",
+        "title": "Azure ML Commitment Plans Management Client",
+        "description": "These APIs allow end users to operate on Azure Machine Learning Commitment Plans resources and their child Commitment Association resources. They support the following operations:<ul><li>Create or update a commitment plan</li><li>Get a commitment plan</li><li>Patch a commitment plan</li><li>Delete a commitment plan</li><li>Get all commitment plans in a resource group</li><li>Get all commitment plans in a subscription</li><li>Get a commitment association</li><li>Get all commitment associations for a commitment plan</li><li>Move a commitment association from one commitment plan to another</li><li>Get usage history for a commitment plan</li></ul>",
+        "version": "2016-05-01-preview"
+    },
+    "host": "management.azure.com",
+    "schemes": [
+        "https"
+    ],
+    "consumes": [
+        "application/json",
+        "text/json"
+    ],
+    "produces": [
+        "application/json",
+        "text/json"
+    ],
+    "paths": {
+        "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearning/commitmentPlans/{planName}/commitmentAssociations/{associationName}": {
+            "get": {
+                "tags": [
+                    "CommitmentAssociation"
+                ],
+                "operationId": "CommitmentAssociation_GetCommitmentAssociation",
```

Comment:
- We follow the NOUN_VERB pattern for operationId. Please take a look over [here](https://github.com/Azure/autorest/wiki/operation-id-noun-in-verb) to get more context behind this approach.
- Make sure the operation Group name (the NOUN in the operationId) is plural. This avoids conflict with model definition that has the same name.